src/controllers/controllerengine.cpp

```diff
@@ -933,35 +928,22 @@ void ControllerEngine::trigger(QString group, QString name) {
    Input:   Script filename
    Output:  false if the script file has errors or doesn't exist
    -------- ------------------------------------------------------ */
-bool ControllerEngine::evaluate(const QString& scriptName, QList<QString> scriptPaths) {
+bool ControllerEngine::evaluate(const QFileInfo& scriptFile) {
     if (m_pEngine == nullptr) {
         return false;
     }
 
-    QString filename = "";
```

Comment:
These changes to ControllerEngine are likely creating conflicts with #1795. Could you take care of those after this is merged? I don't think it'd be fair to ask @ferranpujolcamins to keep fixing the the merge conflicts when that branch has been waiting for a long time.